okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpChannelBuilder.java

```diff
@@ -144,6 +158,19 @@ public OkHttpChannelBuilder connectionSpec(ConnectionSpec connectionSpec) {
   protected ChannelEssentials buildEssentials() {
     final ExecutorService executor = (transportExecutor == null)
         ? SharedResourceHolder.get(DEFAULT_TRANSPORT_THREAD_POOL) : transportExecutor;
+    switch (negotiationType) {
+      case TLS:
+        if (sslSocketFactory == null) {
+          sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+        }
+        break;
+      case PLAINTEXT:
+        sslSocketFactory = null;
```

Comment:
Why bother with this? Why not just set `sslSocketFactory` to its default `if (negotiationType == NegotiationType.TLS && sslSocketFactory == null)`?